[RELEASE] Merge DEVELOP into BETA

rafaelks opened the pull request:
@RocketChat/android

Pancor committed:
adds const

Pancor committed:
add save/get functions

Pancor committed:
add show soft keyboard at activity start functionality

Pancor committed:
refactor code and add clear message possibility

Pancor committed:
add https

Pancor committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

Pancor committed:
add missing bracket

leonardoaramaki committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

Pancor committed:
add support for muliserver

Pancor committed:
code refactor

Pancor committed:
remove unused function

Pancor committed:
remove stateAlwaysHidden

Pancor committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

Pancor committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

leonardoaramaki committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

Pancor committed:
Merge branch 'develop-2.x' into feature/save-unfinished-message

luciofm committed:
Initial message db models and DAO, save messages to database

divyanshub024 committed:
onboarding screen

divyanshub024 committed:
connect server screen

divyanshub024 committed:
Merge branch 'develop' into redesignOnboarding

philipbrito committed:
Merge branch 'develop' into redesignOnboarding

philipbrito committed:
Merge branch 'v3.0.0' into redesignOnboarding

divyanshub024 committed:
login screen

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
new screens

divyanshub024 committed:
Merge remote-tracking branch 'origin/redesignOnboarding' into redesignOnboarding

# Conflicts:
#	build.gradle

divyanshub024 committed:
join community

divyanshub024 committed:
undo

divyanshub024 committed:
new screens

divyanshub024 committed:
onboarding transaltion

divyanshub024 committed:
Merge remote-tracking branch 'origin/develop-2.x' into develop-2.x

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginPresenter.kt
#	app/src/main/java/chat/rocket/android/authentication/login/presentation/LoginView.kt
#	app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt
#	app/src/main/res/layout/fragment_authentication_log_in.xml
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
minor changes

divyanshub024 committed:
onboarding translation

divyanshub024 committed:
login option string translation

divyanshub024 committed:
login string translation

divyanshub024 committed:
sign up string translation

divyanshub024 committed:
2FA string translation

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	app/src/main/java/chat/rocket/android/authentication/login/ui/LoginFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/presentation/AuthenticationNavigator.kt
#	app/src/main/java/chat/rocket/android/authentication/twofactor/ui/TwoFAFragment.kt
#	app/src/main/java/chat/rocket/android/authentication/ui/AuthenticationActivity.kt
#	app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt
#	app/src/main/res/layout/fragment_authentication_log_in.xml
#	app/src/main/res/layout/fragment_profile.xml
#	app/src/main/res/menu/navigation.xml
#	app/src/main/res/values-de/strings.xml
#	app/src/main/res/values-es/strings.xml
#	app/src/main/res/values-fr/strings.xml
#	app/src/main/res/values-hi-rIN/strings.xml
#	app/src/main/res/values-pt-rBR/strings.xml
#	app/src/main/res/values-ru/strings.xml
#	app/src/main/res/values/strings.xml
#	build.gradle

divyanshub024 committed:
Merge branch 'develop-2.x' into redesignOnboarding

# Conflicts:
#	gradle/wrapper/gradle-wrapper.properties

divyanshub024 committed:
resolve conflicts

divyanshub024 committed:
Merge branch 'v3.0.0' into redesignOnboarding

divyanshub024 committed:
update gradle

divyanshub024 committed:
Merge remote-tracking branch 'origin/redesignOnboarding' into redesignOnboarding

divyanshub024 committed:
Add hindi translation

divyanshub024 committed:
update gradle

divyanshub024 committed:
Show menu in the toolbar for "terms and service" and "privacy policy".

leonardoaramaki committed:
Merge branch 'feature/save-unfinished-message' of https://github.com/Pancor/Rocket.Chat.Android into Pancor-feature/save-unfinished-message

philipbrito committed:
Refactor on the OnBoardingFragment

Fix the rocket.chat icon being draw incorrect on on boarding screen.
Adds proper widget themes.
Correct status bar color is now displayed accordingly the fragment (as per our desing).
Removes unneeded code.

leonardoaramaki committed:
In some random, unexpected cases, the emoji keyboard may initialize in an inconsistent state and crash; this prevents the crash

leonardoaramaki committed:
Modularize all suggestions autocompletion code

philipbrito committed:
Updates the sign in to your server view

philipbrito committed:
Merge pull request #1668 from RocketChat/beta

[RELEASE] Update DEVELOP with BETA

leonardoaramaki committed:
Add support for custom emojis on reactions

philipbrito committed:
Merge branch 'v3.0.0' into redesignOnboarding

philipbrito committed:
Removes the alert dialog when changing from https to http protocol

philipbrito committed:
Updates the  ServerFragment

philipbrito committed:
Merge branch 'develop' into fix-random-crash-emoji-keyboard

leonardoaramaki committed:
Bump android gradle plugin to 3.2.0-rc03 and remove kotlin version from root gradle file

leonardoaramaki committed:
Merge branch 'develop' into new/reactions-with-custom-emojis

leonardoaramaki committed:
Set reactionUiModel lambda param to replace misleading implicit 'it'

leonardoaramaki committed:
Merge branch 'new/reactions-with-custom-emojis' of github.com:RocketChat/Rocket.Chat.Android into new/reactions-with-custom-emojis

leonardoaramaki committed:
Merge branch 'develop' into feature/save-unfinished-message

leonardoaramaki committed:
Merge pull request #1674 from RocketChat/new/reactions-with-custom-emojis

[NEW] Add support for custom emojis on reactions

leonardoaramaki committed:
Merge branch 'develop' into feature/save-unfinished-message

leonardoaramaki committed:
Merge branch 'develop' into suggestions-as-a-module

philipbrito committed:
Merge pull request #1675 from RocketChat/suggestions-as-a-module

[REFACTOR] Modularize all suggestions autocompletion code

philipbrito committed:
Merge branch 'develop' into feature/save-unfinished-message

philipbrito committed:
Merge branch 'develop' into fix-random-crash-emoji-keyboard

leonardoaramaki committed:
Implement emoji shortnames autocompletion

philipbrito committed:
Merge pull request #1666 from RocketChat/fix-random-crash-emoji-keyboard

[FIX] Fix random crash emoji keyboard

leonardoaramaki committed:
Merge branch 'develop' into feature/save-unfinished-message

leonardoaramaki committed:
Make EmojiSuggestionsAdapter use TrieCompletionStrategy with a result threshold of 5 items

leonardoaramaki committed:
Show emoji drawable on the left of the suggestions view

rafaelks committed:
Merge branch 'develop' into autocomplete-emojis

luciofm committed:
Fixed merging issues and save attachments

leonardoaramaki committed:
Add collapsable feature to quotes but default to expanded state

leonardoaramaki committed:
Add some missing TODOs on strings

leonardoaramaki committed:
Set view more/less button back to visible again if quoted text size is big enough

leonardoaramaki committed:
Fix markdown parsing for mentions, emphasis and strong emphasis

leonardoaramaki committed:
Merge pull request #958 from Pancor/feature/save-unfinished-message

[NEW] Save unfinished message

leonardoaramaki committed:
Merge branch 'develop' into collpasable-quotes

leonardoaramaki committed:
Merge branch 'develop' into fix-markdown-mentions-and-emphasis

leonardoaramaki committed:
Merge branch 'develop' into autocomplete-emojis

luciofm committed:
Remove unused tests

luciofm committed:
Merge pull request #1690 from RocketChat/autocomplete-emojis

[NEW] Autocomplete emojis

luciofm committed:
Merge branch 'develop' into collpasable-quotes

luciofm committed:
Merge pull request #1695 from RocketChat/collpasable-quotes

[NEW] Add collapsable feature to quotes

luciofm committed:
Merge branch 'develop' into fix-markdown-mentions-and-emphasis

philipbrito committed:
Redesign authentication views.

luciofm committed:
Merge pull request #1697 from RocketChat/fix-markdown-mentions-and-emphasis

[FIX] Fix markdown parsing for mentions, emphasis and strong emphasis

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/RocketChat/Rocket.Chat.Android?pullRequest=1701) <br/>All committers have signed the CLA.

rafaelks commented:
@Pancor Could you please sign the CLA in order to get your commits in the project please?

rafaelks committed:
Merge branch 'beta' into develop

philipbrito committed:
Fix login with deep link and activity lifecycle

philipbrito committed:
Removes unused resources.

philipbrito committed:
Revert "Removes unused resources."

This reverts commit 6dd1ec6e54f7be71903b8cb2ca142d2c30bbdf84.

philipbrito committed:
Removes unsed resources.

Pitstopper committed:
Update strings.xml

Added translation "msg_view_more", "msg_view_less''

philipbrito committed:
Fix expand and collapse state for the accounts view.

philipbrito committed:
GSL is now invoked without needing to tap the key icon.

philipbrito committed:
Merge branch 'v3.0.0' into new/redesign-authentication-views

luciofm committed:
Merge branch 'develop' into feature/db-save-messages

philipbrito committed:
Check for server accounts information before showing the login options view.

This also will show the login with username/email + password with there is only this method enabled by the server.

philipbrito committed:
Add wordpress icon.

philipbrito committed:
Adds scrolling to the server view.

ardabeyazoglu committed:
add turkish language support

leonardoaramaki committed:
Fix the logic to render multiple mentions to the same user and @all or @here

leonardoaramaki committed:
Bump Android gradle plugin to 3.2.0 stable version

leonardoaramaki committed:
Merge pull request #1702 from Pitstopper/patch-9

[I18N] Added translation "msg_view_more", "msg_view_less'' in RU

leonardoaramaki committed:
Merge branch 'develop' into fix-multiple-mentions-to-same-user-not-being-rendered

leonardoaramaki committed:
Merge branch 'develop' into project-bump-android-gradle-plugin

leonardoaramaki committed:
Merge pull request #1713 from RocketChat/project-bump-android-gradle-plugin

[PROJECT] Bump Android gradle plugin to 3.2.0 stable version

leonardoaramaki committed:
Merge branch 'develop' into fix-multiple-mentions-to-same-user-not-being-rendered

leonardoaramaki committed:
Merge branch 'develop' into develop

luciofm committed:
Read messages from DB

leonardoaramaki committed:
Merge pull request #1710 from ardabeyazoglu/develop

[I18N] Add turkish language support

luciofm committed:
Implement Attachments mapping.

luciofm committed:
Remove test code

luciofm committed:
Remove unused code

luciofm committed:
Merge branch 'develop' into feature/db-save-messages

luciofm committed:
Remove unused/missing import

leonardoaramaki committed:
Merge branch 'develop' into fix-multiple-mentions-to-same-user-not-being-rendered

leonardoaramaki committed:
Merge pull request #1712 from RocketChat/fix-multiple-mentions-to-same-user-not-being-rendered

[FIX] Fix the logic to render multiple mentions to the same user

leonardoaramaki committed:
Add room_privacy_changed system message handling

luciofm committed:
Tentative Actions Attachment support on DB

luciofm committed:
Merge branch 'develop' into feature/db-save-messages

luciofm committed:
Fix Status updates

luciofm committed:
Clean up unused code

philipbrito committed:
Updates the width of the spinner.

To make it better on some densities.

philipbrito committed:
Updates views.

philipbrito committed:
Fix the register button state.

philipbrito committed:
Fix bug that wasn't showing the login options for the first time.

philipbrito committed:
Fix getState not encoding correctly.

philipbrito committed:
Merge branch 'v3.0.0' into new/redesign-authentication-views

philipbrito committed:
Fix profile view not showing edit texts.

philipbrito committed:
Disables the GSL feature for now.

 To be improved soon.

luciofm committed:
Fixes on color attachments

rafaelks committed:
Revert "[PROJECT] Bump Android gradle plugin to 3.2.0 stable version"

rafaelks committed:
Merge pull request #1726 from RocketChat/revert-1713-project-bump-android-gradle-plugin

Revert "[PROJECT] Bump Android gradle plugin to 3.2.0 stable version"

rafaelks committed:
Merge branch 'develop' into feature/db-save-messages

rafaelks committed:
Merge branch 'develop' into new/redesign-authentication-views

rafaelks committed:
Merge pull request #1669 from RocketChat/feature/db-save-messages

[FEATURE] Initial message db support

philipbrito committed:
Merge branch 'develop' into new/redesign-authentication-views

philipbrito committed:
Fix the two factor authentication when login with email address.

leonardoaramaki committed:
Add pt-br translation and TODO comments where appropriate

rafaelks committed:
Merge pull request #1716 from RocketChat/new-room-type-changes

[NEW] Add room_privacy_changed system message handling

rafaelks committed:
Merge pull request #1728 from RocketChat/fix/two-fa-with-email

[FIX] Two factor authentication when using email.

philipbrito committed:
Merge branch 'develop' into new/redesign-authentication-views

philipbrito committed:
Add missing strings and remove the unneeded ones.

leonardoaramaki committed:
Fix view less/view more button getting lost

philipbrito committed:
text_email should be disabled.

philipbrito committed:
Updates ServerFragment.kt

philipbrito committed:
Updates the actionmode color.

luciofm committed:
Merge pull request #1700 from RocketChat/new/redesign-authentication-views

[NEW] Redesign authentication views

luciofm committed:
Merge branch 'develop' into fix-collapsing-quote

luciofm committed:
Merge pull request #1729 from RocketChat/fix-collapsing-quote

[FIX] View less/view more button getting lost
